tshirts/TShirts.java

```diff
@@ -0,0 +1,32 @@
+import java.util.InputMismatchException;
+
+class TShirts {
+    static String size(int cms) {
+        if (cms <= 0) {
+            throw new InputMismatchException("T shirt size should be positive and greater than zero!!");
+        }
+        if (cms <= 38) {
+            return "S";
+        } else if (cms <= 42) {
+            return "M";
+        } else if (cms <= 46) {
+            return "L";
+        } else {
+            throw new InputMismatchException("Size is not supported");
+        }
+    }
+
+    public static void main(String[] args) {
+        assert (TShirts.size(37).equals("S"));

```

Comment:
you have exception handling for size  <= 0, and beyond 46.
Good that you have covered all boundary values. But if you are expecting the code to fail when value <=0 is passed then make it explicit in the test case. Similarly for value >=46

Reply:
Okay will  improve that one!